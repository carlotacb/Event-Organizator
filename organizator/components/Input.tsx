import React from "react";
import { View, Text, TextInput, TextInputProps, Platform } from "react-native";
// @ts-ignore
import styled from "styled-components/native";
import FontAwesome from "@expo/vector-icons/FontAwesome";

const Container = styled(View)`
  margin-bottom: 20px;
`;

const TextContainerRow = styled(View)`
  display: flex;
  flex-direction: row;
  gap: 5px;
`;

const TextLabel = styled(Text)`
  margin-bottom: 8px;
  margin-left: 15px;
`;

const TextRequiredLabel = styled(Text)`
  color: red;
`;

const TextError = styled(Text)`
  margin-top: 7px;
  color: red;
  font-size: 12px;
`;

const ErrorIcon = styled(FontAwesome)`
  margin-top: 8px;
  font-size: 12px;
  color: red;
  margin-left: 15px;
`;

const TextInputContainer = styled(View)<{
  isFocus: boolean;
  disabled: boolean;
}>`
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props: { isFocus: boolean; disabled: boolean }) =>
    props.isFocus ? "transparent" : props.disabled ? "#969696" : "#eaeaea"};
  border-radius: 20px;
  border-width: 2px;
  border-color: ${(props: { isFocus: boolean }) =>
    props.isFocus ? "#a9a9a9" : "#eaeaea"};
  padding: 5px 20px;
`;

const InputStyled = styled(TextInput)<{
  disabled: boolean;
}>`
  color: ${(props: { disabled: boolean }) =>
    props.disabled ? "#eaeaea" : "black"};
  flex: 1;
  margin-left: 10px;
  margin-right: 10px;
`;

const InputIcon = styled(FontAwesome)<{
  disabled: boolean;
}>`
  font-size: 20px;
  color: ${(props: { disabled: boolean }) =>
    props.disabled ? "#eaeaea" : "dimgray"};
`;

const EyeIcon = styled(FontAwesome)`
  font-size: 15px;
  color: darkslategray;
`;

interface InputProps extends TextInputProps {
  label: string;
  error?: string;
  iconName?: string | never;
  password?: boolean;
  required?: boolean;
  disabled?: boolean;
}

export default function Input(props: InputProps) {
  const { label, error, required, password, iconName, disabled } = props;

  const [hidePassword, setHidePassword] = React.useState(password);
  const [isFocused, setIsFocused] = React.useState(false);

  return (
    <Container>
      {disabled ? null : (
        <TextContainerRow>
          <TextLabel>{label}</TextLabel>
          <TextRequiredLabel>{required ? "*" : ""}</TextRequiredLabel>
        </TextContainerRow>
      )}
      <TextInputContainer isFocus={isFocused} error={error} disabled={disabled}>
        {iconName && <InputIcon name={iconName} disabled={disabled} />}
        <InputStyled
          disabled={disabled}
          style={[
            Platform.OS === "web"
              ? { outlineStyle: "none", padding: 10 }
              : { padding: 10 },
          ]}
          onFocus={() => {
            if (!disabled) setIsFocused(true);
          }}
          onBlur={() => {
            if (!disabled) setIsFocused(false);
          }}
          secureTextEntry={hidePassword}
          editable={!disabled}
          {...props}
        />
        {password && (
          <EyeIcon
            onPress={() => setHidePassword(!hidePassword)}
            name={hidePassword ? "eye" : "eye-slash"}
          />
        )}
      </TextInputContainer>
      {error && (
        <TextContainerRow>
          <ErrorIcon name="warning" />
          <TextError>{error}</TextError>
        </TextContainerRow>
      )}
    </Container>
  );
}
