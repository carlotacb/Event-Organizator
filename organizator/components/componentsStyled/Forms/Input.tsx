import React from "react";
import { View, TextInput, TextInputProps, Platform } from "react-native";
// @ts-ignore
import styled from "styled-components/native";
import FontAwesome from "@expo/vector-icons/FontAwesome";
import InputLabel from "./InputLabel";
import InputError from "./InputError";

const Container = styled(View)<{ width: string }>`
  margin-bottom: 20px;
  width: ${(props: { width: string }) => props.width};
`;

const TextInputContainer = styled(View)<{
  isFocus: boolean;
  disabled: boolean;
}>`
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props: { isFocus: boolean; disabled: boolean }) =>
    props.isFocus ? "transparent" : props.disabled ? "#969696" : "#eaeaea"};
  border-radius: 15px;
  border-width: 2px;
  border-color: ${(props: { isFocus: boolean }) =>
    props.isFocus ? "#a9a9a9" : "#eaeaea"};
  padding: 3px 20px;
`;

const InputStyled = styled(TextInput)<{
  disabled: boolean;
}>`
  color: ${(props: { disabled: boolean }) =>
    props.disabled ? "#eaeaea" : "black"};
  flex: 1;
  margin-left: 10px;
  margin-right: 10px;
`;

const InputIcon = styled(FontAwesome)<{
  disabled: boolean;
}>`
  font-size: 20px;
  color: ${(props: { disabled: boolean }) =>
    props.disabled ? "#eaeaea" : "dimgray"};
`;

const EyeIcon = styled(FontAwesome)`
  font-size: 15px;
  color: darkslategray;
`;

interface InputProps extends TextInputProps {
  label?: string;
  error?: string;
  iconName?: string | never;
  password?: boolean;
  required?: boolean;
  disabled?: boolean;
  width?: string;
}

export default function Input(props: InputProps) {
  const {
    label,
    error,
    required,
    password,
    iconName,
    disabled,
    width = "auto",
  } = props;

  const [hidePassword, setHidePassword] = React.useState(password);
  const [isFocused, setIsFocused] = React.useState(false);

  return (
    <Container width={width}>
      {disabled && !label ? null : (
        <InputLabel label={label || ""} required={required} />
      )}
      <TextInputContainer isFocus={isFocused} error={error} disabled={disabled}>
        {iconName && <InputIcon name={iconName} disabled={disabled} />}
        <InputStyled
          disabled={disabled}
          style={[
            Platform.OS === "web"
              ? { outlineStyle: "none", padding: 10 }
              : { padding: 10 },
          ]}
          onFocus={() => {
            if (!disabled) setIsFocused(true);
          }}
          onBlur={() => {
            if (!disabled) setIsFocused(false);
          }}
          secureTextEntry={hidePassword}
          editable={!disabled}
          {...props}
        />
        {password && (
          <EyeIcon
            onPress={() => setHidePassword(!hidePassword)}
            name={hidePassword ? "eye" : "eye-slash"}
          />
        )}
      </TextInputContainer>
      {error && <InputError error={error} />}
    </Container>
  );
}
