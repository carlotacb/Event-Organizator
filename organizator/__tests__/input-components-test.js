import * as React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import Input from "../components/componentsStyled/Forms/Input";

it(`renders correctly a input with only label`, () => {
  const tree = renderer.create(<Input label="username" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with label and value`, () => {
  const tree = renderer
    .create(<Input label="username" value="test" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with label and value and error`, () => {
  const tree = renderer
    .create(<Input label="username" value="test" error="error" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with label and value that is password`, () => {
  const tree = renderer
    .create(<Input label="password" value="test" password />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with label and value that is required`, () => {
  const tree = renderer
    .create(<Input label="username" value="test" required />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with label and value that is disabled`, () => {
  const tree = renderer
    .create(<Input label="username" value="test" disabled />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly a input with value and icon`, () => {
  const tree = renderer
    .create(<Input label="username" value="test" icon="user" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
