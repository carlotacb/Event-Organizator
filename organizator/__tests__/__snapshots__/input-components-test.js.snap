// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a input with label and value 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      username
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    />
  </View>
   
  <View
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
  </View>
</View>
`;

exports[`renders correctly a input with label and value and error 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      username
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    />
  </View>
   
  <View
    error="error"
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      error="error"
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
  </View>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "red",
            "fontSize": 12,
            "marginTop": 7,
          },
        ]
      }
    >
      error
    </Text>
  </View>
</View>
`;

exports[`renders correctly a input with label and value that is disabled 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    disabled={true}
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#969696",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      disabled={true}
      editable={false}
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        [
          {
            "color": "#eaeaea",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
  </View>
</View>
`;

exports[`renders correctly a input with label and value that is password 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      password
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    />
  </View>
   
  <View
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      label="password"
      onBlur={[Function]}
      onFocus={[Function]}
      password={true}
      secureTextEntry={true}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "color": "darkslategray",
              "fontSize": 15,
            },
          ],
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ÔÅÆ
    </Text>
  </View>
</View>
`;

exports[`renders correctly a input with label and value that is required 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      username
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    >
      *
    </Text>
  </View>
   
  <View
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      required={true}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
  </View>
</View>
`;

exports[`renders correctly a input with only label 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      username
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    />
  </View>
   
  <View
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly a input with value and icon 1`] = `
<View
  style={
    [
      {
        "marginBottom": 20,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 5,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 8,
            "marginLeft": 15,
          },
        ]
      }
    >
      username
    </Text>
    <Text
      style={
        [
          {
            "color": "red",
          },
        ]
      }
    />
  </View>
   
  <View
    isFocus={false}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#eaeaea",
          "borderColor": "#eaeaea",
          "borderRadius": 20,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 55,
          "paddingBottom": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 5,
        },
      ]
    }
  >
    <TextInput
      editable={true}
      icon="user"
      label="username"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        [
          {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginLeft": 10,
            "marginRight": 10,
          },
          {
            "outlineStyle": "none",
            "padding": 10,
          },
        ]
      }
      value="test"
    />
  </View>
</View>
`;
